src/Elastic.Markdown/IO/Paths.cs

```diff
@@ -18,6 +21,16 @@ private static DirectoryInfo RootDirectoryInfo()
 		return directory ?? new DirectoryInfo(Directory.GetCurrentDirectory());
 	}
 
+	public static IDirectoryInfo? DetermineSourceDirectoryRoot(IDirectoryInfo sourceDirectory)
+	{
+		IDirectoryInfo? sourceRoot = null;
+		var directory = sourceDirectory;
+		while (directory != null && directory.GetDirectories(".git").Length == 0)
```

Comment:
Just wondering if we could somehow use the native `git rev-parse --show-toplevel` function. But I understand it's probably easier this way.